### ISSUE TITLE
[a11y]: Components that utilize the `decorator` prop is not accessible

### ISSUE THREAD

**Issue by AlexanderMelox** [carbon-design-system/carbon#20019]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11.87.1

### React version

19

### Automated testing tool and ruleset

IBM Equal Access

### Description

Currently, passing in a `decorator` child _if_ is interactive like the AILabel's is an a11y concern. Accessibility standards prohibits the use of interactive elements to be passed in to `<label>` or `<legend>`. 

An easy fix for this is to render the `decorator` as a sibling of label and legends, not a child.

<img width="2180" height="640" alt="Image" src="https://github.com/user-attachments/assets/45055ac2-3b81-4c56-99e3-635dbbf0af95" />

We also have to look at every other instance that uses decorators and allow users to not be able to do the following:
```
labelText={
   <>
    Label text
    <Tooltip ... />
   </>
}
```

### Reproduction/example

n/a

### Steps to reproduce

<img width="1310" height="690" alt="Image" src="https://github.com/user-attachments/assets/d566dc6a-d41b-4525-90f7-569ecfa9b733" />
1. Navigate to any component that uses `decorator`
2. The decorator is a child of the label

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by AlexanderMelox** [comment]
We had this issue where passing in a Tooltip as a child of checkbox label will trigger the tooltip and the checkbox to trigger

**Comment by tay1orjones** [comment]
@AlexanderMelox could you reproduce that in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-19-vite?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here?

**Comment by tay1orjones** [comment]
> We also have to look at every other instance that uses decorators and allow users to not be able to do the following:
> 
> ```
> labelText={
>    <>
>     Label text
>     <Tooltip ... />
>    </>
> }
> ```

For this we might be able to repurpose the `useNoInteractiveChildren` to warn users when they pass interactive elements to props that will cause an a11y violation.

**Comment by AlexanderMelox** [comment]
@tay1orjones getting errors on carbon's stackblitz

<img width="625" height="741" alt="Image" src="https://github.com/user-attachments/assets/49e540bd-2dfc-4cae-bd76-da50d4691b4b" />
